agent/src/agent/pipeline/config/handlers/base.py

```diff
@@ -96,6 +96,10 @@ class BaseConfigHandler:
     def __init__(self, pipeline: Pipeline, base_config: dict):
         self.config = base_config
         self.pipeline = pipeline
+        self._check_pipeline()
```

Comment:
maybe we also need to rename this class because base sounds like it fits for everything

Reply:
probably, but it will affect lots of files, also we have another class BaseConfigLoader which in this case also should be renamed. Are you okay with this?